client/extensions/woocommerce/state/sites/shipping-zone-locations/actions.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Internal dependencies
+ */
+import request from '../request';
+import { setError } from '../status/wc-api/actions';
+import {
+	WOOCOMMERCE_SHIPPING_ZONE_LOCATIONS_REQUEST,
+	WOOCOMMERCE_SHIPPING_ZONE_LOCATIONS_REQUEST_SUCCESS,
+} from 'woocommerce/state/action-types';
+import {
+	areShippingZoneLocationsLoaded,
+	areShippingZoneLocationsLoading,
+} from './selectors';
+
+export const fetchShippingZoneLocationsSuccess = ( siteId, zoneId, data ) => {
+	return {
+		type: WOOCOMMERCE_SHIPPING_ZONE_LOCATIONS_REQUEST_SUCCESS,
+		siteId,
+		zoneId,
+		data,
+	};
+};
+
+export const fetchShippingZoneLocations = ( siteId, zoneId ) => ( dispatch, getState ) => {
+	if ( areShippingZoneLocationsLoaded( getState(), zoneId, siteId ) ||
+		areShippingZoneLocationsLoading( getState(), zoneId, siteId ) ) {
+		return;
+	}
+
+	const getAction = {
+		type: WOOCOMMERCE_SHIPPING_ZONE_LOCATIONS_REQUEST,
+		siteId,
+		zoneId,
+	};
+
+	dispatch( getAction );
+
+	return request( siteId ).get( 'shipping/zones/' + zoneId + '/locations' )
+		.then( ( data ) => {
+			dispatch( fetchShippingZoneLocationsSuccess( siteId, zoneId, data ) );
```

Comment:
do you need to break fetchShippingZoneLocationsSuccess into a separate function @ryelle mentioned this on a similar PR I had